### PR TITLE
Sc analysis

### PR DESCRIPTION
**Pull request by andrewram4287**

### CODE REVIEW

**Review by aarmey — CHANGES_REQUESTED** [review]
What does this do? What does it aim to do? Please describe.

**Comment by andrewram4287** [comment]
> What does this do? What does it aim to do? Please describe.

Uploaded all the PopAlign data, filtered genes, performed NNMF, converted from DF to Xarray, ran minimization of loglik for the given data to determine the factors, and also ran cross validation to find the best factor/cluster combination 

Also, added more descrioption to each function

File: gmm/figures/figure7.py [hunk 1]
```diff
@@ -1,34 +1,122 @@
...
+
+# def jax_run():
+#     """Allows JAX to run on GPU"""
+#     # Global flag to set a specific platform, must be used at startup
+#     jax.config.update('jax_platform_name', 'cpu')
+#     x = jnp.square(2)
+#     print(repr(x.device_buffer.device()))  # CpuDevice(id=0)
+
+
+def geneGMM_CV(X, numFolds: int, numClusters: int, numRank: int, maxiter=300):
```
Comment (aarmey) [inline gmm/figures/figure7.py:104]: Is there a reason you're not able to use the existing `tensorGMM_CV()`?

File: gmm/figures/figure7.py [hunk 2]
```diff
@@ -1,34 +1,122 @@
...
+    genesDF, geneNames = import_thompson_drug()
+    genesN = normalizeGenes(genesDF, geneNames)
+    filteredGeneDF, logmean, logstd = mu_sigma(genesDF, geneNames)
+    finalDF, filtered_index = gene_filter(filteredGeneDF, logmean, logstd, offset_value = offset)
+    return finalDF
+
+# def jax_run():
+#     """Allows JAX to run on GPU"""
+#     # Global flag to set a specific platform, must be used at startup
+#     jax.config.update('jax_platform_name', 'cpu')
```
Comment (aarmey) [inline gmm/figures/figure7.py:99]: You can just leave this one line commented out at the top of the file. No need for a whole function.

**Review by aarmey — CHANGES_REQUESTED** [review]
A couple small adjustments.

File: gmm/figures/figure7.py (hunk 1 above)
Comment (andrewram4287) [reply gmm/figures/figure7.py:104]: Nope! Just fixed it!

File: gmm/figures/figure7.py [hunk 3]
```diff
@@ -1,34 +1,100 @@
...
+        },
+    )
+
+    return PopAlignXA
+
+
+def gene_import(offset):
+    """Imports gene data from PopAlign and perfroms gene filtering process"""
+    genesDF, geneNames = import_thompson_drug()
+    genesN = normalizeGenes(genesDF, geneNames)
```
Comment (aarmey) [inline gmm/figures/figure7.py:97]: @andrewram4287 last question—you don't seem to ever use `genesN` here. Is that intended?

**Review by aarmey — APPROVED** [review]

File: gmm/figures/figure7.py (hunk 3 above)
Comment (andrewram4287) [reply gmm/figures/figure7.py:97]: Shoot that is not intended. `mu_sigma(genesDF,geneNames) ` should be `mu_sigma(genesN,geneNames) `. I will run that again and get the correct csv data uploaded with the normalization included.
